src/plugins/data/public/search/session_service.ts

```diff
@@ -30,6 +30,18 @@ export class SessionService implements ISessionService {
   }
   private appChangeSubscription$?: Subscription;
   private curApp?: string;
+  private http?: HttpStart;
+
+  /**
+   * Has the session already been stored (i.e. "sent to background")?
+   */
+  private _isStored: boolean = false;
+
+  /**
+   * Is this session a restored session (have these requests already been made, and we're just
+   * looking to re-use the previous search IDs)?
+   */
+  private _isRestore: boolean = false;
```

Comment:
Note to reviewers: 
We discussed that we are  going to change in follow up to use observable or state containers. 
Let's not focus on this service while reviewing this pr. This changes just needed for showcasing server side service functionality with example plugin